src/js/otus/components/Detail/Isolates/EditIsolate.jsx

```diff
@@ -0,0 +1,49 @@
+import { DialogPortal } from "@radix-ui/react-dialog";
+import { capitalize } from "lodash-es";
+import React from "react";
+import { connect } from "react-redux";
+import { Dialog, DialogContent, DialogOverlay, DialogTitle } from "../../../../base";
+import { editIsolate, hideOTUModal } from "../../../actions";
+import IsolateForm from "./IsolateForm";
+
+function EditIsolate(props) {
+    function handleSubmit({ sourceName, sourceType }) {
+        props.onSave(props.otuId, props.isolateId, sourceType, sourceName);
+    }
+
+    return (
+        <Dialog open={props.show} onOpenChange={props.onHide}>
+            <DialogPortal>
+                <DialogOverlay />
+                <DialogContent>
+                    <DialogTitle>Edit Isolate</DialogTitle>
+                    <IsolateForm
+                        sourceType={capitalize(props.sourceType)}
+                        sourceName={props.sourceName}
+                        allowedSourceTypes={props.allowedSourceTypes}
+                        restrictSourceTypes={props.restrictSourceTypes}
+                        onSubmit={handleSubmit}
+                    />
+                </DialogContent>
+            </DialogPortal>
+        </Dialog>
+    );
+}
```

Comment:
I feel like complete conversion would be in scope for this PR. Not blocking, will merge either way. Also happy with an issue made and merging as is